### PR TITLE
Migrate to PF4, Upgrade to Webpack 5, Enable EL9 builds

### PR DESCRIPTION
**Pull request by sjd78**
## Summary
The project has been completely migrated to PF4.  PF3 is no longer used.  Webpack has been upgrade to version 5.  Github actions for PR CI and for main branch builds now include EL9 builds.

_This PR is a condensed version of and supersedes:_
  - https://github.com/oVirt/ovirt-engine-ui-extensions/pull/65
  - https://github.com/oVirt/ovirt-engine-ui-extensions/pull/68
  - https://github.com/oVirt/ovirt-engine-ui-extensions/pull/69

Bug-Url: https://bugzilla.redhat.com/show_bug.cgi?id=2021123

## Migrate modal Spinners to PF4

With these changes, all modals are now fully PF4:
  - Migrate the PF3 `Spinner` use to `LoadingSpinner` that uses a PF4 `Spinner`
  - Remove PF3 CSS imports from the `plugin.js` modal base page
  - Every modal that supports fake data now has a sleep to be able to see the loading spinner

## Migrate Dashboard to PF4, remove PF3, align CSS

_Note: These changes remove all use of PF3, Bootstrap and jQuery from the project enabling the upgrade to webpack 5 and EL9 builds to follow._

### Screen shots

Fake data dashboard:
screenshot-engine-dev ovirt_8080-2022 07 06-10_13_40

Fake data dashboard loading:
screenshot-engine-dev ovirt_8080-2022 07 06-10_14_58

HeatMap tooltip:
screenshot-engine-dev ovirt_8080-2022 07 06-10_16_01

Utilization popups:
screenshot-engine-dev ovirt_8080-2022 07 06-10_21_59

### Dashboard changes

Project dependencies changes:
  - remove patternfly (v3),
  - remove patternfly-react (v3),
  - and resolutions that are no longer needed

PF refactoring on dashboard:
  - Dashboard imports now are consistent with only PF4 imports, PF3 imports have been removed

  - The initial data loading spinner and the loading error message are converted to PF4 and render properly in the horizontal and vertical center of the iframe page

  - Refactor `RefreshDataControls` from pf3 to pf4-react

  - Upgrade `HeatMap` to a functional component and use PF4 Tooltips:
    - Convert `HeatMap` to a functional component
    - Drop use of bootstrap and jquery for Tooltips
    - Drop use of D3 for rendering the `HeatMap`'s SVG
    - Adopt use of PF4 Tooltips attached to every SVG `rect` generated

  - Handle layout changes needed after switching to the pf4 reset css

The base CSS used by plugin, modals (loaded from the plugin iframe), and the dashboard page are now aligned.  They all use the same base `plugin-pf4-overrides.css` to make the pages look closer to the PF3 styles of webadmin.

### Underlying changes
  - Upgrade PF4 packages to 2022.07 release
    Closest previous release: 2020.13 (2020-10-06)
    New Release: 2022.07 (2022-06-03)
    React Notes: https://www.patternfly.org/v4/developer-resources/release-notes#2022.07-release-notes-2022-06-02
    HTML Notes: https://www.patternfly.org/v4/developer-resources/release-notes/html#2022.07-release-notes-2022-06-03

  - Upgrade `css-element-queries` used in `HeightMatching` from 0.4.0 to 1.2.3

  - Remove c3, jquery and bootstrap, HeatMap will not need them

  - Drop some unnecessary jest polyfills

  - Remove logger.js, rely on fancy-console babel plugin

  - DashboardDataProvider use fake-data sleep to show the loading spinner

  - To aid in visual testing, always render the `MonitoringPortalLink` on dashboard when `config.useFakeData` is enabled.  If graphana is not configured on the dev engine, the url is rendered with a simple javascript console log.
 

## Upgrade to Webpack 5
  - The build is now running on Webpack 5

  - All webpack dependencies have been updated to the current versions compatible with WP5

  - Old and unused loaders and plugins have been removed

  - The `@buildinfo` banner is added to all generated JS and CSS and is retained in the production builds.  Removed "Change-Id" tracking since the project is no longer in gerrit.

  - Font inlining to CSS is simplified and updated following the current `patternfly-react-seed` project webpack configurations

  - CSS imports on the entrypoint JS files refactored to be the first items imported.  They should be imported before the application components.  See [@patternfly/react-core#project-configuration](https://github.com/patternfly/patternfly-react/tree/main/packages/react-core#project-configuration).

  - HeatMap refactored to use `d3-scale` instead of `d3`.  This reduces the size of the vendor chunk considerably.
 
  - Drop unnecessary `glob-parent` resolution:
    - No (runtime) dependencies require `glob-parent`
    - All devDependencies references at least the version in the resolution line

##  Enable EL9 builds
  - Add EL9 builds to github actions using matrix definitions
 
  - `ovirt-engine-nodejs-modules` will be installed or upgraded as needed

  - CI will run 2 jobs for each platform (EL8 and EL9):
    - an offline build using `ovirt-engine-nodejs-modules` (the results of this job are saved as exported artifacts)
    - an online build using `yarn install` directly (the results are discarded)

### CODE REVIEW

File: static/css/dashboard.css [hunk 1]
```diff
@@ -22,9 +12,18 @@ body, .containers-dashboard {
...
+}
+
+#app {
+  min-height: 100%;
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
 }
 
 #global-dashboard {
```
Comment (rszwajko) [inline static/css/dashboard.css:29]: Adding `width: 100%` here solved problems with resizing for me. With this the charts and heat maps fit the space. The only thing that is triggering overflow are the top cards (data centers, hosts, etc) but this is due to explicit `flex-wrap: nowrap` and seems to be this way before (not a regression).

**Review by rszwajko — CHANGES_REQUESTED** [review]
Overall looks good - only the resizing needs a fix as the content is then hidden due to overflow (see my suggestion below)

File: static/css/dashboard.css (hunk 1 above)
Comment (sjd78) [reply static/css/dashboard.css:29]: For getting narrow or wide?  I was playing with the responsive view and things look decent....

oVirt Open Virtualization Manager - Google Chrome_068

oVirt Open Virtualization Manager - Google Chrome_069

and wide looks ok as well...
oVirt Open Virtualization Manager - Google Chrome_070

Comment (rszwajko) [reply static/css/dashboard.css:29]: https://user-images.githubusercontent.com/64194103/178750018-01bb739c-d01b-4c80-8759-0794c9c7f2c1.mp4

Comment (rszwajko) [reply static/css/dashboard.css:29]: On my env it looked differently - it's Chromium, dev mode with fake data.

Comment (sjd78) [reply static/css/dashboard.css:29]: Ok so at least on Chromium, setting a `max-width: 100%` on the `#global-dishboard` very much helps.  It doesn't hurt the other browsers either.

Going down to a browser width of 970px with the left nav collapsed to icons still does not require horizontal scrolling. That covers most devices in landscape mode down to phones.

Changing the inventory status card to layout different under that size was always a point of contention.  There may be some reasonable ways to handle small widths.  I'll experiment a little bit.

Comment (sjd78) [reply static/css/dashboard.css:29]: Looking good now:

[screencast-engine-dev.ovirt_8443-2022.07.14-21_49_34.webm](https://user-images.githubusercontent.com/3985964/179130757-97213355-7103-44fd-bce0-ea4a4c551025.webm)

**Review by rszwajko — APPROVED** [review]
The new reactive layout looks great!

**Review by sgratch — APPROVED** [review]
LGTM

**Comment by isaranova** [comment]
LGTM

**Pull request merged**